### PR TITLE
Import tests from std library - part 5/16

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Part 5/16
Please see details in part 1/16 - https://github.com/oneapi-src/oneDPL/pull/885
This PR should be merged into main only after https://github.com/oneapi-src/oneDPL/pull/885

### CODE REVIEW

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.gtequal/test_rev_iter_op_gteq.pass.cpp [hunk 1]
```diff
@@ -68,30 +66,29 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.gtequal/test_rev_iter_op_gteq.pass.cpp:90]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.lt/test_rev_iter_op_lt.pass.cpp [hunk 2]
```diff
@@ -68,30 +66,29 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.lt/test_rev_iter_op_lt.pass.cpp:90]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ltequal/test_rev_iter_op_lteq.pass.cpp [hunk 3]
```diff
@@ -68,30 +66,29 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ltequal/test_rev_iter_op_lteq.pass.cpp:90]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.minus/difference_type_reverse_iter_op_minus.pass.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.minus/difference_type_reverse_iter_op_minus.pass.cpp:82]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ne/test_rev_iter_op_ne.pass.cpp [hunk 5]
```diff
@@ -69,30 +67,29 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ne/test_rev_iter_op_ne.pass.cpp:91]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.plus/difference_type_rev_iter_op_plus.pass.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.plus/difference_type_rev_iter_op_plus.pass.cpp:82]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ref/op_arrow_rev_iter_op_ref.pass.cpp [hunk 7]
```diff
@@ -144,12 +139,12 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ref/op_arrow_rev_iter_op_ref.pass.cpp:146]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.star/op_star_rev_iter.pass.cpp [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    return ret;
+}
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.star/op_star_rev_iter.pass.cpp:93]: it makes sense to add more details in the message.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.sum/difference_type_reverse_iter_op_sum.pass.cpp [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.sum/difference_type_reverse_iter_op_sum.pass.cpp:83]: it makes sense to add more details in the message.

**Review by MikeDvorskiy — COMMENTED** [review]
Looks good excepting wording for the error messages.

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.cons/reverse_iter_cons.pass.cpp [hunk 10]
```diff
@@ -54,41 +52,40 @@ struct Derived : Base
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.cons/reverse_iter_cons.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.compound.assign.plus/difference_type_reverse_iter_op_comp_assign_plus.pass.cpp [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.compound.assign.plus/difference_type_reverse_iter_op_comp_assign_plus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbequal/test_rev_iter_db_eq.pass.cpp [hunk 12]
```diff
@@ -69,31 +67,29 @@ kernel_test()
...
-main(int, char**)
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbequal/test_rev_iter_db_eq.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbminus/post_rev_iter_op_dbminus.pass.cpp [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbminus/post_rev_iter_op_dbminus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ne/test_rev_iter_op_ne.pass.cpp (hunk 5 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ne/test_rev_iter_op_ne.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.plus/difference_type_rev_iter_op_plus.pass.cpp (hunk 6 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.plus/difference_type_rev_iter_op_plus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ref/op_arrow_rev_iter_op_ref.pass.cpp (hunk 7 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ref/op_arrow_rev_iter_op_ref.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.star/op_star_rev_iter.pass.cpp (hunk 8 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.star/op_star_rev_iter.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.sum/difference_type_reverse_iter_op_sum.pass.cpp (hunk 9 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.sum/difference_type_reverse_iter_op_sum.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.compound.assign.minus/difference_type_rev_iter_op_comp_assign_minus.pass.cpp [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.compound.assign.minus/difference_type_rev_iter_op_comp_assign_minus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbminus/pre_rev_iter_op_dbminus.pass.cpp [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbminus/pre_rev_iter_op_dbminus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbplus/post_rev_iter_op_dbplus.pass.cpp [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.dbplus/post_rev_iter_op_dbplus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.diff/test_ref_iter_op_diff.pass.cpp [hunk 17]
```diff
@@ -65,29 +63,28 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.diff/test_ref_iter_op_diff.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.equal/reverse_iter_op_eq.pass.cpp [hunk 18]
```diff
@@ -58,43 +56,42 @@ struct Derived : Base
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.equal/reverse_iter_op_eq.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.great/test_rev_iter_op_gt.pass.cpp [hunk 19]
```diff
@@ -68,30 +66,29 @@ kernel_test()
...
+main()
 {
+#ifdef TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.great/test_rev_iter_op_gt.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.index/difference_type_rev_iter_op_index.pass.cpp [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#ifdef TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of work in kernel_test()");
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.index/difference_type_rev_iter_op_index.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.lt/test_rev_iter_op_lt.pass.cpp (hunk 2 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.lt/test_rev_iter_op_lt.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ltequal/test_rev_iter_op_lteq.pass.cpp (hunk 3 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.ltequal/test_rev_iter_op_lteq.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.minus/difference_type_reverse_iter_op_minus.pass.cpp (hunk 4 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.minus/difference_type_reverse_iter_op_minus.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.gtequal/test_rev_iter_op_gteq.pass.cpp (hunk 1 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/reverse.iterators/reverse.iter.ops/reverse.iter.op.gtequal/test_rev_iter_op_gteq.pass.cpp]: Fixed

**Review by MikeDvorskiy — APPROVED** [review]
LGTM

**Pull request merged**